sct.c

```diff
@@ -26,108 +29,151 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <math.h>
 
-static void usage()
+static void usage(char * pname)
 {
-    printf("Usage: xsct [temperature]\n"
-        "Temperatures must be in a range from 1000-10000\n"
-        "If the argument is 0, xsct resets the display to the default temperature (6500K)\n"
-        "If no arguments are passed, xsct estimates the current display temperature\n"
-        "If -h or --help is passed xsct will display this usage information\n");
+    printf("Xsct (1.5)\n"
+        "Usage: %s [options] [temperature]\n"
+        "\tIf the argument is 0, xsct resets the display to the default temperature (6500K)\n"
+        "\tIf no arguments are passed, xsct estimates the current display temperature\n"
+        "Options:\n"
+        "\t-v, --verbose \t xsct will display debug information\n"
+        "\t-h, --help \t xsct will display this usage information\n", pname);
 }
 
-/* cribbed from redshift, but truncated with 500K steps */
-static const struct { float r; float g; float b; } whitepoints[] = {
-    { 1.00000000,  0.18172716,  0.00000000, }, /* 1000K */
-    { 1.00000000,  0.42322816,  0.00000000, },
-    { 1.00000000,  0.54360078,  0.08679949, },
-    { 1.00000000,  0.64373109,  0.28819679, },
-    { 1.00000000,  0.71976951,  0.42860152, },
-    { 1.00000000,  0.77987699,  0.54642268, },
-    { 1.00000000,  0.82854786,  0.64816570, },
-    { 1.00000000,  0.86860704,  0.73688797, },
-    { 1.00000000,  0.90198230,  0.81465502, },
-    { 1.00000000,  0.93853986,  0.88130458, },
-    { 1.00000000,  0.97107439,  0.94305985, },
-    { 1.00000000,  1.00000000,  1.00000000, }, /* 6500K */
-    { 0.95160805,  0.96983355,  1.00000000, },
-    { 0.91194747,  0.94470005,  1.00000000, },
-    { 0.87906581,  0.92357340,  1.00000000, },
-    { 0.85139976,  0.90559011,  1.00000000, },
-    { 0.82782969,  0.89011714,  1.00000000, },
-    { 0.80753191,  0.87667891,  1.00000000, },
-    { 0.78988728,  0.86491137,  1.00000000, }, /* 10000K */
-    { 0.77442176,  0.85453121,  1.00000000, },
-};
-#define TEMPERATURE_DEFAULT 6500
-#define TEMPERATURE_MIN     1000
-#define TEMPERATURE_MAX     10000
-#define TEMPERATURE_STEP    500
+#define TEMPERATURE_NORM    6500
+#define TEMPERATURE_ZERO    700
 #define GAMMA_MULT          65535.0
-#define AVG(c,temp,ratio) whitepoints[(temp) / TEMPERATURE_STEP].c * (1 - (ratio)) + whitepoints[(temp) / TEMPERATURE_STEP + 1].c * (ratio)
+#define GAMMA_K0GR          -1.47751309139817
```

Comment:
a comment in the code referencing the source of these constants would be good

Reply:
Resolve https://github.com/faf0/sct/commit/0de52379321c80e58a2fe082d84c77fbd8a3d744